### PR TITLE
Test module for random Value generation

### PR DESCRIPTION
**Pull request by pontusmelke**
replaces https://github.com/neo4j/neo4j/pull/11707

### CODE REVIEW

File: community/random-values/src/main/java/org/neo4j/values/Generator.java [hunk 1]
```diff
@@ -19,17 +19,48 @@
...
 
+    /**
+     * Return a pseudorandom normally distributed long between 0 (inclusive) and the given bound(exlusive)
+     * @param bound the exclusive upper bound for the number generation
+     * @return a pseudorandom normally distributed int
+     */
     int nextInt( int bound );
 
+    /**
+     * Return a pseudorandom normally distributed float
```
Comment (fickludd) [inline community/random-values/src/main/java/org/neo4j/values/Generator.java]: In what range?

**Review by fickludd — APPROVED** [review]

**Review by tinwelint — CHANGES_REQUESTED** [review]
I'm going to put my "Request changes" mark on this too, just as the last one just for the record. It's mostly about the RandomValues' relationship with RandomRule, which we talked about offline.

**Comment by pontusmelke** [comment]
@tinwelint I have changed so that `RandomRule` and `RandomValues` get along better now.

@tinwelint @fickludd @SaschaPeukert I think I have adressed all of your comments from the old PR.

File: community/index/pom.xml [hunk 2]
```diff
@@ -88,5 +88,10 @@ the relevant Commercial Agreement.
       <groupId>org.mockito</groupId>
       <artifactId>mockito-core</artifactId>
     </dependency>
+    <dependency>
+      <groupId>org.neo4j</groupId>
+      <artifactId>neo4j-random-values</artifactId>
+      <version>${project.version}</version>
```
Comment (tinwelint) [inline community/index/pom.xml]: this is a test dependency, no?

**Review by tinwelint — APPROVED** [review]
Looks awesome to me! Only had this comment that random-values dependencies should have scope=test, shouldn't they?

File: community/index/pom.xml (hunk 2 above)
Comment (pontusmelke) [reply community/index/pom.xml]: yes, nice catch!

**Review by fickludd — APPROVED** [review]
Awesome, let's merge this.

**Pull request merged**